react/features/calendar-sync/middleware.js

```diff
@@ -262,27 +262,40 @@ function _parseCalendarEntry(event, knownDomains) {
  * @returns {void}
  */
 function _updateCalendarEntries(events) {
-    if (events && events.length) {
-        // eslint-disable-next-line no-invalid-this
-        const { dispatch, getState } = this;
+    if (!events || !events.length) {
+        return;
+    }
 
-        const knownDomains = getState()['features/base/known-domains'];
-        const eventList = [];
+    // eslint-disable-next-line no-invalid-this
+    const { dispatch, getState } = this;
 
-        const now = Date.now();
+    const knownDomains = getState()['features/base/known-domains'];
+    const now = Date.now();
+    const entryMap = new Map();
 
-        for (const event of events) {
-            const calendarEntry = _parseCalendarEntry(event, knownDomains);
+    for (const event of events) {
+        const entry = _parseCalendarEntry(event, knownDomains);
 
-            calendarEntry
-                && calendarEntry.endDate > now
-                && eventList.push(calendarEntry);
-        }
+        // The function _parseCalendarEntry may return null if it couldn't parse
+        // a calendar entry that is suitable for displaying.
+        if (entry && entry.endDate > now) {
+            const { id } = entry;
+            const existingEntry = entryMap.get(id);
 
-        dispatch(
-            setCalendarEvents(
-                eventList
-                    .sort((a, b) => a.startDate - b.startDate)
-                    .slice(0, MAX_LIST_LENGTH)));
+            // Recurring calendar entries have the same ID. Since we only want
```

Comment:
This PR is not achieving any of the 2 goals that I described! I tested on Pixel 2 with Android P and the Calendar entries have different IDs even though the event is recurring. I'm not even sure why we're deduplicating by ID. When I was describing the task I said:

We're showing certain pieces of information such as URL, date and time, etc. If the pieces of information that we display are the same, then as far as the user can tell, the entries are the same.

@zbettenbuk, may I please ask you to try harder!